cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/preview/PreviewDatasetFramework.java

```diff
@@ -56,97 +92,138 @@ public PreviewDatasetFramework(DatasetFramework local, DatasetFramework actual,
+      localDatasetFramework.updateInstance(datasetInstanceId, props);
+    }
   }
 
   @Override
   public Collection<DatasetSpecificationSummary> getInstances(NamespaceId namespaceId)
     throws DatasetManagementException {
-    return null;
+    return localDatasetFramework.getInstances(namespaceId);
   }
 
   @Nullable
   @Override
   public DatasetSpecification getDatasetSpec(DatasetId datasetInstanceId) throws DatasetManagementException {
-    return null;
+    if (datasetNames.contains(datasetInstanceId.getDataset())) {
+      return actualDatasetFramework.getDatasetSpec(datasetInstanceId);
+    }
+    return localDatasetFramework.getDatasetSpec(datasetInstanceId);
   }
 
   @Override
   public boolean hasInstance(DatasetId datasetInstanceId) throws DatasetManagementException {
-    return false;
+    if (datasetNames.contains(datasetInstanceId.getDataset())) {
+      return actualDatasetFramework.hasInstance(datasetInstanceId);
+    }
+    return localDatasetFramework.hasInstance(datasetInstanceId);
   }
 
   @Override
   public boolean hasSystemType(String typeName) throws DatasetManagementException {
-    return false;
+    return localDatasetFramework.hasSystemType(typeName);
   }
 
   @Override
   public boolean hasType(DatasetTypeId datasetTypeId) throws DatasetManagementException {
-    return false;
+    return localDatasetFramework.hasType(datasetTypeId);
   }
 
   @Nullable
   @Override
   public DatasetTypeMeta getTypeInfo(DatasetTypeId datasetTypeId) throws DatasetManagementException {
-    return null;
+    return localDatasetFramework.getTypeInfo(datasetTypeId);
   }
 
   @Override
   public void truncateInstance(DatasetId datasetInstanceId) throws DatasetManagementException, IOException {
+    // If dataset exists in the preview space then only truncate it otherwise its a no-op
+    if (localDatasetFramework.hasInstance(datasetInstanceId)) {
+      localDatasetFramework.truncateInstance(datasetInstanceId);
+    }
   }
 
   @Override
   public void deleteInstance(DatasetId datasetInstanceId) throws DatasetManagementException, IOException {
+    // If dataset exists in the preview space then only delete it otherwise its a no-op
+    if (localDatasetFramework.hasInstance(datasetInstanceId)) {
+      localDatasetFramework.deleteInstance(datasetInstanceId);
+    }
   }
 
   @Override
   public void deleteAllInstances(NamespaceId namespaceId) throws DatasetManagementException, IOException {
+    localDatasetFramework.deleteAllInstances(namespaceId);
   }
 
   @Nullable
   @Override
+  @SuppressWarnings("unchecked")
   public <T extends DatasetAdmin> T getAdmin(DatasetId datasetInstanceId, @Nullable ClassLoader classLoader)
     throws DatasetManagementException, IOException {
-    return null;
+    // Return the no-op admin for the dataset from the real space
+    if (datasetNames.contains(datasetInstanceId.getDataset())
+      && actualDatasetFramework.hasInstance(datasetInstanceId)) {
+      return (T) NOOP_DATASET_ADMIN;
```

Comment:
The reason why it is `<T extends DatasetAdmin>` is because the admin instance is created by `DatasetDefinition`, which can be any sub-type of `DatasetAdmin`. If the call knows what the type is, he can cast it (implicitly by Java) so that he can call specific method on that sub-type.